clients/drcachesim/simulator/cache_simulator.cpp

```diff
@@ -59,14 +60,17 @@ cache_simulator_t::cache_simulator_t(const cache_simulator_knobs_t &knobs_) :
                 knobs_.warmup_fraction, knobs_.sim_refs,
                 knobs_.cpu_scheduling, knobs_.verbose),
     knobs(knobs_),
-    icaches(NULL),
-    dcaches(NULL),
-    is_warmed_up(false)
+    l1_icaches(NULL),
+    l1_dcaches(NULL),
+    is_warmed_up(false),
+    is_two_level(true)
 {
     // XXX i#1703: get defaults from hardware being run on.
 
-    llcache = create_cache(knobs.replace_policy);
-    if (llcache == NULL) {
+    // This configuration allows for one shared LLC only.
+    cache_t *llc = new cache_t;
```

Comment:
Looks like a leak

Reply:
You assign llc to `new cache_t` here and on the next line throw that away and assign llc to `create_cache` which does its own `new cache_t`.  This duplicate new is incorrect, a leak, and should be removed.

Reply:
You are right. Fixed.